OpenMPI and OpenMP Makefiles for Archer2

VladCarare opened the pull request:
The old instructions for Archer2 needed updating. 

The pull also contains instructions for the new MPI implementation of GAP on Archer2. 

This was tested and worked as of today.

VladCarare committed:
Updated Makefile.archer2 to abide by new Archer2

VladCarare committed:
Create Makefile.archer2_openmpi+openmp

VladCarare committed:
Added note for Archer2

VladCarare committed:
Added note for Archer2

Sideboard commented on arch/Makefile.archer2_openmpi+openmp:
```diff
@@ -0,0 +1,55 @@
+LINKER = ftn
+
+# OpenMP
+DEFINES += -D_OPENMP
```
You can remove this line (41) to get rid of `Warning: "_OPENMP" redefined`. Most compilers add this macro if you select the openmp option, that's why the warning is given when you declare it again manually.

VladCarare replied:
Thanks! Modified

VladCarare committed:
Omit OPENMP redefinition

Deleted 
`DEFINES += -D_OPENMP` line to get rid of OPENMP redefined warning. 

https://github.com/libAtoms/QUIP/pull/424#discussion_r845872799

VladCarare committed:
Omit OPENMP redefinition

Deleted 
`DEFINES += -D_OPENMP` line to get rid of OPENMP redefined warning. 

https://github.com/libAtoms/QUIP/pull/424#discussion_r845872799

Sideboard commented on arch/Makefile.archer2_openmpi+openmp:
```diff
@@ -0,0 +1,54 @@
+
+# declarations
+
+include arch/Makefile.linux_x86_64_gfortran
```
If you include the Makefile.archer2 here (30) instead you can remove all duplicate declarations and only keep the mpi specific ones.

VladCarare replied:
Indeed I agree. I was trying to be consistent with the other makefiles in this repo ([Makefile.linux_x86_64_gfortran_openmpi+openmp](https://github.com/libAtoms/QUIP/blob/public/arch/Makefile.linux_x86_64_gfortran_openmpi%2Bopenmp)).

I'll modify as you said

VladCarare committed:
Slim file

Call include arch/Makefile.archer2 which includes all other definitions needed

jameskermode commented:
thanks @VladCarare and @Sideboard
